File: src/main/java/br/com/gabrielalmir/gotcha/utils/Base62Converter.java
package br.com.gabrielalmir.gotcha.utils;

public class Base62Converter {
    private static final String BASE62 = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";

    public static String toBase62(long value) {
        StringBuilder sb = new StringBuilder();
        while (value > 0) {
            int remainder = (int) (value % 62);
            sb.append(BASE62.charAt(remainder));
            value /= 62;
        }
        return sb.reverse().toString();
    }
}

File: src/utils/base62converter.ts
export class Base62Converter {
    private static BASE62 = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';

    public static toBase62(value: number): string {
        if (value === 0) {
            return '0';
        }

        let sb = '';
        while (value > 0) {
            const remainder = value % 62;
            sb = Base62Converter.BASE62.charAt(remainder) + sb;
            value = Math.floor(value / 62);
        }
        return sb;
    }
}
